src/python/pants/backend/scala/lint/scalafmt/rules.py

```diff
@@ -233,14 +227,13 @@ async def setup_scalafmt(
             os.path.join(source_dir, name) for name in files_in_source_dir
         )
 
-    jdk_invoke_args = jdk_setup.args(bash, tool_classpath.classpath_entries(toolcp_relpath))
     partitions = await MultiGet(
         Get(
             Partition,
             SetupScalafmtPartition(
+                classpath_entries=tuple(tool_classpath.classpath_entries(toolcp_relpath)),
                 merged_sources_digest=merged_sources_digest,
-                immutable_input_digests=FrozenDict(immutable_input_digests),
-                jdk_invoke_args=jdk_invoke_args,
+                extra_immutable_input_digests=FrozenDict(extra_immutable_input_digests),
```

Comment:
The wrapping `FrozenDict` here shouldn't be necessary anymore I think.

Also, I think that these need to be nailgun keys as well: they are the classpath of the tool itself.